triggers/daml/AutoApproval.daml

```diff
@@ -167,7 +167,11 @@ generateCommands choiceArg queryResults = map (\(cid, _) -> (toAnyContractId cid
 generateCommandsWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (t2 -> t -> Bool) -> (ContractId t2 -> c) -> [(ContractId t2, t2)] -> [(ContractId t, t)] -> [(AnyContractId, Command)]
 generateCommandsWithCid _ _ [] _ = []
 generateCommandsWithCid _ _ _ [] = []
-generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> (\(cid, _) -> (toAnyContractId cid, exerciseCmd cid $ toChoice reqCid)) <$> findFirstMatch req targets pred) requests
+generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> getCommandWithCid toChoice reqCid <$> findFirstMatch req targets pred) requests
+
+-- | Takes a choice and a contract id which is a parameter of that choice, the contract which exposes this choice and builds up a command
+getCommandWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (ContractId t2 -> c) -> ContractId t2 -> (ContractId t, t) -> (AnyContractId, Command)
```

Comment:
Do you need all these type constraints on here? e.g. `Choice` constraint? I recall we removed it from a similar function.

Reply:
Only when I remove the entire type signature for the function, it works. 

Removing the `choice` constraint causes `Could not deduce (HasExercise t c r0) arising from a use of ‘exerciseCmd’` error. Also, removing the type constraint while leaving the rest of the signature causes an error.